torchao/quantization/pt2e/quantize_pt2e.py

```diff
@@ -217,6 +235,32 @@ def _quant_node_constraint(n: Node) -> bool:
     return n.op == "call_function" and n.target in _QUANT_OPS
 
 
+def _is_torchao_prepared_do_not_use_outside_this_file(model):
+    from torchao.quantization.pt2e.fake_quantize import (
+        FakeQuantize as torchao_FakeQuantize,
+    )
+    from torchao.quantization.pt2e.observer import ObserverBase as torchao_ObserverBase
+
+    is_torch_ao_prepared = False
+    is_torchao_prepared = False
+    for _, m in model.named_modules():
+        if isinstance(
+            m, torch.ao.quantization.fake_quantize.FakeQuantize
+        ) or isinstance(m, torch.ao.quantization.observer.ObserverBase):
+            is_torch_ao_prepared = True
+        if isinstance(m, torchao_FakeQuantize) or isinstance(m, torchao_ObserverBase):
+            is_torchao_prepared = True
+    assert is_torch_ao_prepared or is_torchao_prepared, (
+        "Must be prepared using torch.ao or torchao, but did not see any FakeQuantize or ObserverBase modules in model: "
+        + str([f"{n}: {type(m)}" for n, m in model.named_modules()])
+    )
```

Comment:
there might be models that does not have observer/fake_quant inserted, I think we could remove this assert